### PR TITLE
feat: add Object Detection with Cloud Vision API

### PR DESCRIPTION
**Pull request by jauntybrain**
Object Detection Extension Icon

Adds [Object Detection](https://cloud.google.com/vision/docs/object-localizer) Extension, powered by Cloud Vision API.
The extension is fully built out and tested.

### CODE REVIEW

File: .DS_Store
Comment (pr-Mais) [inline .DS_Store:1]: Could you remove all .DS_Store files

**Comment by jauntybrain** [comment]
@cabljac fixed linting issues, should pass now
@pr-Mais removed .DS_Store files

**Comment by dackers86** [comment]
Hi @jauntybrain 

This looks a fantastic new extension, thanks for the PR!

As this is a new extension, and not updates a new one - how do you feel about owning this even publishing under your own profile and repository?

I'm sure we can find a home for it, it's just about deciding if you would like to take care of the ownership and maintenance. 

Let us know what you think!

**Comment by jauntybrain** [comment]
Hey @dackers86

Thank you for the feedback! I will gladly host the extension in my repository and take full care of ownership/maintenance. 

Does this mean I will be able to publish this extension on **[extensions.dev](https://extensions.dev/)**?

**Comment by dackers86** [comment]
Yes, of course.

Please find the relevant link below. 

https://firebase.google.com/docs/extensions/publishers/register
https://firebase.google.com/docs/extensions/publishers/upload-and-publish

I'll close this PR for now, but feel free to comment if you have any questions!

**Pull request closed without merging**